swap_meet/vendor.py

```diff
@@ -1,2 +1,139 @@
+
+
 class Vendor:
-    pass
\ No newline at end of file
+    """
+    Represents a vendor with a list of inventory items
+    """
+    def __init__(self,inventory = None):
+        """
+        instatiates Vendor instances; has attribute 'inventory', which
+        initialized as an empty list if no value is passed in.
+        """
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self,item):
+        """
+        Adds item to a vendor's inventory list
+        """
+        self.inventory.append(item)
+        return item
+
+    def remove(self,item):
+        """
+        Removes an item from a vendor's inventory list if the item is
+        present and returns the item; otherwise returns False
+        """
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    def get_by_id(self, id):
+        """
+        Returns an item by ID number; returns None if no item matches
+        the provided ID number
+        """
+        for item in self.inventory:
+            if id is item.id:
+                return item
+
+        return None
+
+    def swap_items(self, other_Vendor, my_item, their_item):
+        """
+        Swaps given item from one vendor's inventory list with an item
+        from another vendor's inventory list and returns True. If
+        either list is missing the given item, method returns False
+        """
+        if my_item in self.inventory and their_item in other_Vendor.inventory:
+            self.inventory.remove(my_item)
+            other_Vendor.inventory.append(my_item)
+
+            other_Vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+
+            return True
+        else:
+            return False
```

Comment:
Same idea as your `remove()` function! For this particular case, not finding an item in one of the lists is essentially the edge case, while swapping the items successfully is the nominal case! When this happens, feel free to switch your conditional to return False on a negative result and use the rest of the function to run the nominal case. As before, this also removes the nested O(n) efficiencies we see with `remove` and `in`!

```suggestion
        if my_item not in self.inventory or their_item not in other_Vendor.inventory:
            return False
            
        self.inventory.remove(my_item)
        other_Vendor.inventory.append(my_item)

        other_Vendor.inventory.remove(their_item)
        self.inventory.append(their_item)
            
```